### PR TITLE
fix: theme analytics not getting reported as user is being fetched delayed

### PR DESCRIPTION
**Pull request by vikrantgupta25**
### Summary

- the user properties being fetched by API call were delayed hence the event was not getting fired 

#### Screenshots

image

### CODE REVIEW

File: frontend/src/AppRoutes/index.tsx [hunk 1]
```diff
@@ -178,23 +178,25 @@ function App(): JSX.Element {
...
-				LOCALSTORAGE.THEME_ANALYTICS,
-			);
-			if (!isThemeAnalyticsSent) {
-				trackEvent('Theme Analytics', {
-					theme: isDarkMode ? THEME_MODE.DARK : THEME_MODE.LIGHT,
-					user: pick(user, ['email', 'userId', 'name']),
-					org,
-				});
-				setLocalStorageApi(LOCALSTORAGE.THEME_ANALYTICS, 'true');
+		if (user && user?.email && user?.userId && user?.name) {
```
Comment (vikrantgupta25) [inline frontend/src/AppRoutes/index.tsx:181]: added the check to send the event only after the user is fetched. and set the local state to not send further events

File: frontend/src/constants/localStorage.ts [hunk 2]
```diff
@@ -18,5 +18,5 @@ export enum LOCALSTORAGE {
 	DASHBOARD_VARIABLES = 'DASHBOARD_VARIABLES',
 	SHOW_EXPLORER_TOOLBAR = 'SHOW_EXPLORER_TOOLBAR',
 	PINNED_ATTRIBUTES = 'PINNED_ATTRIBUTES',
-	THEME_ANALYTICS = 'THEME_ANALYTICS',
+	THEME_ANALYTICS_V1 = 'THEME_ANALYTICS_V1',
```
Comment (vikrantgupta25) [inline frontend/src/constants/localStorage.ts:21]: renamed the local storage key as older keys would have already been set for the user.

**Review by YounixM — APPROVED** [review]

File: frontend/src/AppRoutes/index.tsx [hunk 3]
```diff
@@ -178,23 +178,25 @@ function App(): JSX.Element {
...
+			try {
+				const isThemeAnalyticsSent = getLocalStorageApi(
+					LOCALSTORAGE.THEME_ANALYTICS_V1,
+				);
+				if (!isThemeAnalyticsSent) {
+					trackEvent('Theme Analytics', {
+						theme: isDarkMode ? THEME_MODE.DARK : THEME_MODE.LIGHT,
```
Comment (SagarRajput-7) [inline frontend/src/AppRoutes/index.tsx:188]: We wanted to move away from trackEvent and use logEvent 

cc: @makeavish

Comment (SagarRajput-7) [reply frontend/src/AppRoutes/index.tsx:188]: This is a NIT for now, approved the PR

**Review by SagarRajput-7 — APPROVED** [review]

File: frontend/src/AppRoutes/index.tsx (hunk 3 above)
Comment (vikrantgupta25) [reply frontend/src/AppRoutes/index.tsx:188]: we will anyways remove this event in a while so should be okay for now

**Pull request merged**